mypy/checker.py

```diff
@@ -4004,9 +4004,15 @@ def is_typed_callable(c: Optional[Type]) -> bool:
 
 
 def is_untyped_decorator(typ: Optional[Type]) -> bool:
-    if not typ or not isinstance(typ, CallableType):
+    if not typ:
         return True
-    return typ.implicit
+    if isinstance(typ, CallableType):
+        return typ.implicit
+    if isinstance(typ, Instance):
+        method = typ.type.get_method('__call__')
+        if method and method.type:
+            return not is_typed_callable(method.type)
+    return True
```

Comment:
If an `Instance` doesn't have a `__call__` method this will return `True`, which will generate an additional spurious error. While I believe it shouldn't.

Could you please add a test case for this?